tests/test_abstract_repr.py

```diff
@@ -1602,29 +1612,52 @@ def test_deserialize_eom_ops(self, correct_phase_drift):
                     "correct_phase_drift": correct_phase_drift,
                 },
             ],
-            variables={"duration": {"type": "int", "value": [100]}},
+            variables={
+                "duration": {"type": "int", "value": [100]},
+                "detuning_on": {"type": "int", "value": [0.0]},
+            },
             device=json.loads(IroiseMVP.to_abstract_repr()),
             channels={"global": "rydberg_global"},
         )
         if correct_phase_drift is None:
             for op in s["operations"]:
                 del op["correct_phase_drift"]
-        _check_roundtrip(s)
+
         seq = Sequence.from_abstract_repr(json.dumps(s))
-        # init + declare_channel + enable_eom_mode
-        assert len(seq._calls) == 3
-        # add_eom_pulse + disable_eom
-        assert len(seq._to_build_calls) == 2
+        # init + declare_channel + enable_eom_mode (if not var_detuning_on)
+        assert len(seq._calls) == 3 - var_detuning_on
```

Comment:
Why is there anything inside _calls if var_detuning_on ? Since the first operation (enable_eom_mode) is parametrized, it is stored in _to_build_calls, but afterwards the next operation should be added to _to_build_calls instead of _calls ?

Reply:
Thanks ! I had not noticed ! All good on my side 👍